ponys.js

```diff
@@ -57,7 +59,10 @@ export default class {
 					this.import(options.name, options.src, options):
 					this.define(options.name, template, options);
 			})
-		);
+		).then(list => {
+			list?.filter(e => e.status !== "fulfilled")
+				.forEach(e => console.warn("import/define failed", e));
+		});
```

Comment:
I'd rather allow the user to handle the Promise(s) for themselves, as that's how they know when all the async calls and definitions are done! And in most cases where the individual Promises will reject (imported file not found, syntax error in the setup script, custom elemenet redefinition, etc.), I think the circumstances will result in an error in the console in any case - and probably with a more specific error message.

Reply:
This was mostly done as a result of the `customElements.get(name)` check I added in this PR, which I had noticed would NOT report in console, and instead would just silently error. I added the logic here as I wanted something that would catch and report on all failures, not just select ones.

As far as an existing silent error, here's one ([fiddle](https://jsfiddle.net/8zw5n60b/1/)):

```html
<script type="module">
  import Ponys from "ponys.js"
  console.log(Ponys.defineAll());
</script>

<!-- no errors -->
<template name="hello-world">
  <div>hello world</div>
</template>

<!-- already defined: silent errors (check promise results) -->
<template name="hello-world">
  <div>hello world 2</div>
</template>

<!-- 404: browser does show a console error in this case -->
<template name="test-component" src="test-component.html"></template>
```

1) Redefining a component with same name results in a silent error
2) Creating a template with an invalid link (404 or otherwise) will indeed produce a visible console error

I suspect there are other cases we're missing, hence the approach I took in this PR.

I'm wondering if our main misunderstanding here is how `Promise.allSettled()` works. It doesn't care if our promises succeed or fail, it just cares about competition. It will only return if all promises have fulfilled. See [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled#description) for a description on the comparison between `Promise.allSettled()` and `Promise.all()`.

Ultimately, this means the following code will fail silently, where the later will fail visibly (from console):

```js
// Silently fails
Promise.allSettled([Promise.reject(1), Promise.reject(2), Promise.reject(3), Promise.resolve(true)])

// Visibly fails
Promise.all([Promise.reject(1), Promise.reject(2), Promise.reject(3), Promise.resolve(true)])
```

Since `Promise.all()` stops the execution chain if there is a failure, but `Promise.allSettled()` continues it regardless, I do think `allSettled` is the correct choice for this library, but we need to know when there are failures! Which was the point of my change. Not to wrap the promise results or catch any specific errors, just to put some console warnings for the user to observe. I do recognize that some of these console warnings may be redundant or displayed multiple times, but I'd rather see multiple than not at all.

We could just leave it up to the user to dig through the returned promise object and find the errors, but I believe that would elude to a bad user experience. I actually experienced it, which is why I went as far as wanting to create a PR to clear this up 🙃.

Hopefully this explanation helps to clear things up!

Reply:
Yes, thank you - but just to check, are you still getting silent errors on the redefinition of `hello-world` via the `<template>` tags if you let the browser throw at the bottom of `define()`, rather than testing for it at the top? :thinking:

Reply:
Yes. Please check the fiddle. Its using the original source code, not my modified version.

Reply:
Ah, yes, now I see: `Promise.allSettled()` is implicitly catching all the errors produced by each individual Promise, but not re-throwing them... bad. In this case, I'd like to forward the original `Error` as follows:
```suggestion
		).then(results => results
			.forEach(result => result.reason && console.error(result.reason));
		});
```